HL7/ig-template-base
# Add TestScript layout and links to Profile(s)

### What does this Pull Request do?
Add components that incorporates TestScript resources to the FHIR Artifacts page, adds a new FHIR Profile page navigation tab "TestScripts" if TestScript resources are found linked to the FHIR Profile, and add a new TestScript layout page.

### Why?
The FHIR community has been increasingly interested in provided better support for testing artifacts within IGs. These additional components are an initial enhancement that allows IGs generated from the HL7 IG template to incorporate and list TestScript resources as first class artifacts and with their associated FHIR Profiles (StructureDefinitions).

### How?
New and updated components:
* **./config.json** 
  * "**extraTemplates**" list - add new object for { "name": "testscripts", "description": "TestScripts" }
  * "**defaults**" list - update "StructureDefinition" object to add new attributes: "template-testscripts", "testscripts"; add new "TestScript" object
* **./includes/fragment-profile-navtabs.html**
  * New **testscript_count** assignment
  * New **TestScripts** navigation tab logic - displays the new "TestScripts" nav tab only if **testscript_count** > 0
* **./layouts/layout-profile-testscripts.html**
  * New layout for TestScript resources, modeled after "layout-profile-examples.html"
* **./scripts/groupings.txt**
  * Add new **grouping** element "-test-testscript" - for FHIR Artifacts page, section "Testing Artifacts: TestScripts"
* **./scripts/onGenerate.greoup.xslt**
  * Add new "**groupingId**" variable assignment "-test-testscript" when resource is a 'TestScript' - works in conjunction with **groupings.txt** new **grouping** element (see previous bullet)

### Dependencies
These modifications are dependent on the IG Publisher tool modifications in the Pull Request https://github.com/HL7/fhir-ig-publisher/pull/518
